src/server/scripts/Northrend/ChamberOfAspects/ObsidianSanctum/boss_sartharion.cpp

```diff
@@ -190,377 +205,415 @@ class boss_sartharion : public CreatureScript
-                    events.RepeatEvent(25000);
-                    return;
-                case EVENT_SARTHARION_START_LAVA:
-                    SendLavaWaves(true);
-                    
-                    return;
-                case EVENT_SARTHARION_FINISH_LAVA:
-                    SendLavaWaves(false);
-                    
-                    return;
+                        SummonLavaWaves();
+                        extraEvents.RepeatEvent(25000);
+                        return;
+                    }
+                    case EVENT_SARTHARION_START_LAVA:
+                    {
+                        SendLavaWaves(true);
+                        return;
+                    }
+                    case EVENT_SARTHARION_FINISH_LAVA:
+                    {
+                        SendLavaWaves(false);
+                        return;
+                    }
+                    // Handling of Drakes Events
+                    // Dragon Calls
+                    case EVENT_SARTHARION_CALL_TENEBRON:
+                    {
+                        Talk(SAY_SARTHARION_CALL_TENEBRON);
+                        if (Creature* tenebron = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_TENEBRON)))
+                        {
+                            tenebron->AI()->DoAction(ACTION_CALL_DRAGON);
+                        }
+                        break;
+                    }
+                    case EVENT_SARTHARION_CALL_SHADRON:
+                    {
+                        Talk(SAY_SARTHARION_CALL_SHADRON);
+                        if (Creature* shadron = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_SHADRON)))
+                        {
+                            shadron->AI()->DoAction(ACTION_CALL_DRAGON);
+                        }
+                        break;
+                    }
+                    case EVENT_SARTHARION_CALL_VESPERON:
+                    {
+                        Talk(SAY_SARTHARION_CALL_VESPERON);
+                        if (Creature* vesperon = ObjectAccessor::GetCreature(*me, instance->GetData64(DATA_VESPERON)))
+                        {
+                            vesperon->AI()->DoAction(ACTION_CALL_DRAGON);
+                        }
+
+                        break;
+                    }
+                }
             }
 
+            // Handle Sartharion combat abbilities
+            events.Update(diff);
+
             if (me->HasUnitState(UNIT_STATE_CASTING))
+            {
                 return;
+            }
 
-            HandleSartharionAbilities();
-            HandleDrakeAbilities();
+            while (uint32 const eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_SARTHARION_CAST_CLEAVE:
+                    {
+                        DoCastVictim(SPELL_SARTHARION_CLEAVE, false);
+                        events.RepeatEvent(10000);
+                    }break;
```

Comment:
why this codestyle? `}break;` ? why not either avoiding braces or keeping the `break` inside the braces? is there any specific reason?

Reply:
well, its my codestyle which i used to be sure that i not missed to add break after code block since in huge code block "base operator" can be misswatched. I'm commonly using this in javascript and php. Will correct in next commit